web-bundle/src/main/java/com/graphhopper/http/GraphHopperManaged.java

```diff
@@ -122,6 +123,15 @@ public GraphHopperManaged(GraphHopperConfig configuration, ObjectMapper objectMa
         }
         configuration.setProfiles(newProfiles);
 
+        // Error if gtfs_link_mapper profile wasn't included in GH config (link mapper step will fail in this case)
```

Comment:
could we do this validation somewhere that's more relevant to GTFS link mapping so it's not as hidden? could go in `GtfsLinkMapper`, or maybe `GtfsLinkMapperCommand` or the `CustomGraphHopperGtfs` constructor if we wanna fail faster. with that approach i think you could remove the outer `if (configuration.has("gtfs.file"))` check

Reply:
I'd kind of argue that the check being here isn't really "hidden", because this class is essentially the entrypoint that runs for any of our graphhopper command-line commands, and I'm hoping to have the check apply to _any_ command that runs where GTFS is relevant (I want the failure to be thrown as early as possible during building -ie, during `import`- when we know the link mapper would fail in the future if we were to continue the router building process).

But I agree that the extra `if (configuration.blah...)` clause isn't great. I guess I could even move it up to the existing `if (configuration.has("gtfs.file"))` block, but I'm fine plopping it into `CustomGraphHopperGtfs` to make things extra clear for others - the behavior there should be the same.